src/Calculator/Views/TitleBar.xaml.cpp

```diff
@@ -167,4 +175,88 @@ namespace CalculatorApp
         VisualStateManager::GoToState(
             this, e->WindowActivationState == CoreWindowActivationState::Deactivated ? WindowNotFocused->Name : WindowFocused->Name, false);
     }
+
+    void TitleBar::AlwaysOnTopButtonClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+    {
+        auto avm = safe_cast<CalculatorApp::ViewModel::ApplicationViewModel ^>(this->DataContext);
+        if (ApplicationView::GetForCurrentView()->GetForCurrentView()->ViewMode == ApplicationViewMode::CompactOverlay)
+        {
+            avm->Mode = m_beforeAlwaysOnTopMode;
+            if (avm->Mode == CalculatorApp::Common::ViewMode::Standard)
+            {
+                avm->CalculatorViewModel->AreHistoryShortcutsEnabled = true;
+                avm->CalculatorViewModel->HistoryVM->AreHistoryShortcutsEnabled = true;
+            }
+            IAsyncOperation<bool> ^ result = ApplicationView::GetForCurrentView()->TryEnterViewModeAsync(ApplicationViewMode::Default);
+            auto changeModeTask = Concurrency::create_task(result);
+            changeModeTask.then([this](bool success) {
+                auto avm = safe_cast<CalculatorApp::ViewModel::ApplicationViewModel ^>(this->DataContext);
+                avm->CalculatorViewModel->IsAlwaysOnTop = !success;
+                if (!success)
+                {
+                    AlwaysOnTopButton->HorizontalAlignment = ::HorizontalAlignment::Left;
+                    AlwaysOnTopButton->Content = L"\uEE47";
+                    AppName->Text = "";
+                    ToolTipService::SetToolTip(AlwaysOnTopButton, "Exit always-on-top");
+                }
+                else
+                {
+                    AlwaysOnTopButton->HorizontalAlignment = ::HorizontalAlignment::Right;
+                    AlwaysOnTopButton->Content = L"\uEE49";
+                    AppName->Text = AppResourceProvider::GetInstance().GetResourceString(L"AppName");
+                    ToolTipService::SetToolTip(AlwaysOnTopButton, "Always on top");
+                }
+                this->Focus(::FocusState::Programmatic);
+            });
+        }
+        else
+        {
+            m_beforeAlwaysOnTopMode = avm->Mode;
+            avm->Mode = CalculatorApp::Common::ViewMode::Standard;
+            avm->CalculatorViewModel->AreHistoryShortcutsEnabled = false;
+            avm->CalculatorViewModel->HistoryVM->AreHistoryShortcutsEnabled = false;
+
+            Windows::Storage::ApplicationDataContainer ^ localSettings = Windows::Storage::ApplicationData::Current->LocalSettings;
+            ViewModePreferences ^ compactOptions = ViewModePreferences::CreateDefault(ApplicationViewMode::CompactOverlay);
+            if (!localSettings->Values->GetView()->HasKey("calculatorAlwaysOnTopLaunched"))
```

Comment:
You should probably create some constant strings for all the keys you use ("calculatorAlwaysOnTopLaunched", "calculatorAlwaysOnTopLastWidth", ...)